togglebutton.js

```diff
@@ -0,0 +1,77 @@
+const template = document.createElement("template");
+template.innerHTML = `
+<style>
+.slider {
+    position: absolute;
+    width: 20px;
+    height: 20px;
+    background-color: #808080;
+    border-radius: 50%;
+    transition-duration: 1s;
+    pointer-events: none;
+  }
+  
+  .container {
+    width: 40px;
+    height: 20px;
+    background-color: black;
+    border-radius: 30% / 50%;
+    position: relative;
+    pointer-events: none;
+  }
+  
+  .slide {
+    transform: translateX(100%);
+  }
+  
+  .container input {
+    opacity: 0;
+    position: absolute;
+    pointer-events: all;
+  }
+  </style>
+            <div class="container">
+            <input  type="checkbox"  id="something"/>
+              <div
+                class="slider"
+                
+              ></div>
+            </div>
+`;
+class ToggleButton extends HTMLElement {
+  static get observedAttributes() {
+    return ["on"];
+  }
+  constructor() {
+    super();
+    this.attachShadow({ mode: "open" });
+    this.shadowRoot.appendChild(template.content.cloneNode(true));
+  }
+
+  connectedCallback() {
+    const radiobutton = this.shadowRoot.querySelector(".slider");
+    if (this.hasAttribute("on")) {
+      radiobutton.classList.add("slide");
+    }
+    this.shadowRoot
+      .querySelector("input")
+      .addEventListener("click", () => this.toggleButton());
```

Comment:
You'll need to make sure that the event listener used in `addEventListener`/`removeEventListener` is the same function. Otherwise, you won't be able to remove it. In this line, for example, an anonymous function is created and used as the event listener.